### ISSUE TITLE
About reproduce Anime style?

### ISSUE THREAD

**Issue by zhongtao93** [williamyang1991/DualStyleGAN#19]
Hi, thanks for sharing code.
I download the Anime dataset according to the README.md, but I get bad result like this:
![image](https://user-images.githubusercontent.com/15263824/165317764-f42b07e6-5585-4720-84d1-c5c3625c526b.png)

Can you provide more details about trainning Anime dataset.
Here are some of my training results, is that correct？I used your provided pretrain model [generator-pretrain.pt](https://drive.google.com/file/d/1j8sIvQZYW5rZ0v1SDMn2VEJFqfRjMW3f/view?usp=sharing) so I skip stage I&II, and fintune dualstylegan(stage III) using parameters as peper:
![image](https://user-images.githubusercontent.com/15263824/165320053-85ecfc6f-62f2-4cda-8c99-98c875fb5500.png)
![image](https://user-images.githubusercontent.com/15263824/165321347-1f43072f-7de9-46e5-88d3-c0678505663c.png)
The picture above is fintune-004800.png，I use one GPU, so I trained 8*600=4800 with batch=4
![image](https://user-images.githubusercontent.com/15263824/165318890-9a04312b-6f39-4111-abe3-acb4f51c2df2.png)
The picture above is destylization log picture.
![image](https://user-images.githubusercontent.com/15263824/165319371-687a17ea-c435-4f30-8bb9-b968c0131c0d.png)
The picture above is dualstylegan-002000.jpg
Hoping for hearing from you soon.

**Comment by williamyang1991** [comment]
Thank you for your interests. 

## Dataset

> Hi, thanks for sharing code. I download the Anime dataset according to the README.md, but I get bad result like this: ![image](https://user-images.githubusercontent.com/15263824/165317764-f42b07e6-5585-4720-84d1-c5c3625c526b.png)

This style image is not in the Anime dataset I trained my model. Did you add new images in the Anime dataset?

## Batch Size and Iterations

> [image] The picture above is fintune-004800.png，I use one GPU, so I trained 8*600=4800 with batch=4 

Obviously, the network is over fine-tuned. You should use much fewer iterations, so that the original real faces can be still recognized. 
Another problem is that it is hard to train StyleGAN/DualStyleGAN with a small batch size. 
Here is a comparion between batch=4\*1=4 and batch=4\*8=32

![comparison](https://user-images.githubusercontent.com/18130694/165325556-0c52968d-57a8-45b7-9fbd-0e135844e502.jpg)

You see, even I did not over fine-tune the StyleGAN, I still get poor results with a single GPU.

## Destylzation 

> ![image](https://user-images.githubusercontent.com/15263824/165318890-9a04312b-6f39-4111-abe3-acb4f51c2df2.png) The picture above is destylization log picture. 

Since your StyleGAN is over fine-tuned, the destylization results have few content correspondences with the input anime faces. 
Here is my result:

![_10005620](https://user-images.githubusercontent.com/18130694/165326688-f34bc441-ad59-49fb-8c81-1d6eaf6bf8ae.png)


## Finetune DualStyleGAN

> [image] The picture above is dualstylegan-002000.jpg 

I also found that finetuning on Anime dataset is the most unstable one among the finetunings on all datasets due to the large discrapancy between real faces and anime faces. And on a single GPU, it becomes much more unstable. I use the following settings to obtain the performance above. 

![image](https://user-images.githubusercontent.com/18130694/165327353-e64cdeeb-4340-4ed3-9c7e-bd5ecd4e6758.png)

On a single GPU, the model often falled into mode collapse so I trained many times and used the best one. Maybe you could use a lower learning rate and try different hyperparameters to find a relatively more stable ones.

**Comment by zhongtao93** [comment]
Thank you for detailed and patient reply～
Yes, I add serveral custom picture, so I total use 189 pictures.
Have you published the paper on arXiv about the single GPU analysis you mentioned above?  Could you please share me the detailed version with the information you mentioned above :). 
I'll try to train with 4 GPUs next, since you let me know the importance of multi GPU～
Thank you again for your reply~

**Comment by williamyang1991** [comment]
This single GPU analysis is in the supplementary material of our submission. 
We remove it in the arxiv version to make the paper more concise. 
Here is the original version:

![batchsize](https://user-images.githubusercontent.com/18130694/165428281-e817c9d0-d1f1-402f-8373-827cd65acc3f.jpg)

**Comment by JuncFang-git** [comment]
> Thank you for detailed and patient reply～ Yes, I add serveral custom picture, so I total use 189 pictures. Have you published the paper on arXiv about the single GPU analysis you mentioned above? Could you please share me the detailed version with the information you mentioned above :). I'll try to train with 4 GPUs next, since you let me know the importance of multi GPU～ Thank you again for your reply~

I am also trained dualstlegan  at 1 V100 GPU with a style which have large discrapancy between real faces. And it overfiting with my style dataset.  It is better after you train at 4GPU?

**Comment by williamyang1991** [comment]
The official stylegan is trained on 4 batch * 8 GPUs. 
Therefore, I think it is better to use this setting to achieve the best performance.
Obviously 4 GPUs are better than 1 GPU, and maybe you need to try different hyperparameters to find a relatively more stable ones on 4GPUs.

And also, you can save many checkpoints and choose the one just before overfiting.

**Comment by JuncFang-git** [comment]
> 



> The official stylegan is trained on 4 batch * 8 GPUs. Therefore, I think it is better to use this setting to achieve the best performance. Obviously 4 GPUs are better than 1 GPU, and maybe you need to try different hyperparameters to find a relatively more stable ones on 4GPUs.
> 
> And also, you can save many checkpoints and choose the one just before overfiting.

Thank you for your reply.
I think my stylegan2 is not over fine-tuning, but a bad result still got when do destyle. Is the style too difficult?
fine-tuned stylegan2:
![image](https://user-images.githubusercontent.com/76929740/178675036-6af83940-3b2c-4270-a369-15cda6e17d30.png)
some destyle results
![image](https://user-images.githubusercontent.com/76929740/178675207-bdf01f3a-fbb5-4960-97bc-582afadb7e66.png)

**Comment by williamyang1991** [comment]
I think the style is too abstract and different from real humans, thus the destylization results are not very ideal.
I have two solutions:
1. From the visual quality, I think you can just use the results in the second column as the destylization results for training dualstylegan. The quality is better than the final destylization results in the last column. 
2. Decrease the epoches for finetuning stylegan. Try different epoches and find a epoch that the finetuned stylegan has the best destylization results.

**Comment by JuncFang-git** [comment]
> I think the style is too abstract and different from real humans, thus the destylization results are not very ideal. I have two solutions:
> 
> 1. From the visual quality, I think you can just use the results in the second column as the destylization results for training dualstylegan. The quality is better than the final destylization results in the last column.
> 2. Decrease the epoches for finetuning stylegan. Try different epoches and find a epoch that the finetuned stylegan has the best destylization results.

Thanks. In fact I have paired dataset, so i will try to get instyle/exstyle code from paired data. If still not work, maybe the style is not suitable for dualstylegan cause it's abstraction.

**Comment by weiweiecho** [comment]
@JuncFang-git hi,do you get a nice result by getting style code from paired data?

**Comment by JuncFang-git** [comment]
> @JuncFang-git hi,do you get a nice result by getting style code from paired data?

No, I can't get a nice result.
I try to train the dualstylegan with paired instyle-exstyle code(both of them get from the psp encoder).  Some checkpoint records are shown bleow: 
0-iter
![image](https://user-images.githubusercontent.com/76929740/181145471-06eedf40-ff6c-4df0-a39d-34f13f03e8ab.png)
5500-iter
![image](https://user-images.githubusercontent.com/76929740/181145546-d8f041d6-53bb-428c-b130-6f6b4f77795c.png)
Then i test the model with some user photo. Some result are shown bleow:
![image](https://user-images.githubusercontent.com/76929740/181145832-2987f459-33bd-42cf-a15c-912155c28f30.png)
![image](https://user-images.githubusercontent.com/76929740/181145855-1bd041d7-7b08-4276-9e43-841221453cd3.png)

**Comment by williamyang1991** [comment]
@JuncFang-git 
Your results are interesting. 
It is quite strange that the model under 5500-iter produces both cartoon faces and real faces.
I haven't seen this phenomenon before.

**Comment by JuncFang-git** [comment]
> @JuncFang-git Your results are interesting. It is quite strange that the model under 5500-iter produces both cartoon faces and real faces. I haven't seen this phenomenon before.

Hahaha, It's interesting.  The left part of DualStyleGAN is a complete psp structure after i used real-image's instyle code. Maybe it's more difficult to change the style by interpolate exstyle code. But i don't know why it produces both cartoon faces and real faces.

**Comment by onefish51** [comment]
> 

@JuncFang-git Hi,I found your style very ”かわいい“，it's funny！The paired data you created by JOJOGAN ？why not train it by an end-to-end net ?

**Comment by onefish51** [comment]
> > 
> 
> > The official stylegan is trained on 4 batch * 8 GPUs. Therefore, I think it is better to use this setting to achieve the best performance. Obviously 4 GPUs are better than 1 GPU, and maybe you need to try different hyperparameters to find a relatively more stable ones on 4GPUs.
> > And also, you can save many checkpoints and choose the one just before overfiting.
> 
> Thank you for your reply. I think my stylegan2 is not over fine-tuning, but a bad result still got when do destyle. Is the style too difficult? fine-tuned stylegan2: ![image](https://user-images.githubusercontent.com/76929740/178675036-6af83940-3b2c-4270-a369-15cda6e17d30.png) some destyle results ![image](https://user-images.githubusercontent.com/76929740/178675207-bdf01f3a-fbb5-4960-97bc-582afadb7e66.png)

@JuncFang-git Hi,I found your style very ”かわいい“，it's funny！The paired data you created by JOJOGAN ？why not train it by an end-to-end net ?
